bots/tests-policy

```diff
@@ -0,0 +1,360 @@
+#!/usr/bin/python2
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2013 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import datetime
+import fnmatch
+import re
+import os
+import socket
+import sys
+import traceback
+
+sys.dont_write_bytecode = True
+
+from task import github
+
+BOTS = os.path.join(os.path.dirname(__file__))
+
+def main():
+    script = os.path.basename(__file__)
+    parser = argparse.ArgumentParser(description='Check a traceback for a known issue')
+    parser.add_argument('-o', "--offline", action='store_true',
+            help="Work offline, don't fetch new data from origin for rebase")
+    parser.add_argument("--simple", action="store_true",
+            default=(script == "image-naughty"),
+            help="Just print out a known issue number")
+    parser.add_argument('context', help="The image to check against")
+    opts = parser.parse_args()
+
+    base = None
+    if os.environ.get("GITHUB_KNOWN_ISSUE_BASE"):
+        netloc = os.environ.get("GITHUB_API", "https://api.github.com")
+        base = "{0}/repos/{1}/".format(netloc, os.environ.get("GITHUB_KNOWN_ISSUE_BASE"))
+
+    api = None if opts.offline else github.GitHub(base=base)
+
+    context = opts.context
+    if "/" not in context:
+        context = "verify/{0}".format(context)
+    prefix, unused, image = context.rpartition("/")
+
+    try:
+        output = sys.stdin.read()
+        number = 0
+
+        if image and output:
+            number = check_known_issue(api, output, image)
+
+        if number and api and api.available:
+            try:
+                post_github(api, number, output, image)
+            except socket.error:
+                traceback.print_exc()
+                sys.stderr.write("{0}: posting update to GitHub failed\n".format(script))
+                # Fall through
+
+        # Simple single number output
+        if opts.simple:
+            if number:
+                sys.stdout.write("{0}\n".format(number))
+            return 0
```

Comment:
AFAICS this will skip `checkRetry()` in "simple" mode, which is a backwards compatibility break (for back branches) as this was previously handled in testlib.py.

Reply:
For back branches, the retry logic is still in the caller. So I believe this should work as expected. At least that's how I reasoned about it.

Reply:
Oh right, as we don't copy testlib.py for testing backbranches. So should be ok.